### PR TITLE
Imp/oo based color checker

### PR DESCRIPTION
**Pull request by lfb0801**
Heb een Object Oriented solution voor discernColor gemaakt.

### CODE REVIEW

File: VanHalenSort/VanHalenSort.ino [hunk 1]
```diff
@@ -23,136 +24,122 @@ MQTTClient client;
...
+    {
+      return ranges[i].name;
+    }
+    return "NONE"
+  }
 };
 
 /////////////////TAKING AN AVERAGE FROM 5 MEASUREMENTS TO SMOOTH THE FREQ RESPONSE OUT
-int[] averageFreqRGB(){
+int[] averageFreqRGB()
```
Comment (lfb0801) [inline VanHalenSort/VanHalenSort.ino:110]: Deze functie hoort naar mijn idee op een andere thread te draaien

File: VanHalenSort/VanHalenSort.ino [hunk 2]
```diff
@@ -23,136 +24,122 @@ MQTTClient client;
...
-      case RED: return "Red";
-      case GREEN: return "Green";
-      case PURPLE: return "Purple";
-      case YELLOW: return "Yellow";
-      case ORANGE: return "Orange";
-   }
+  R = round(R / 5);
+  G = round(G / 5);
+  B = round(B / 5);
+  return int sumRGB[3] = {R, G, B};
```
Comment (lfb0801) [inline VanHalenSort/VanHalenSort.ino:134]: zou een Color object moeten returnen, maar zie überhaupt niet waar het gebruikt wordt.

Comment (slatan1983) [reply VanHalenSort/VanHalenSort.ino:134]: nope. this is just the read out from the sensor (averaged over 5 consecuitive reads). this still has to be compared to the ranges to determine the color.

**Review by slatan1983 — COMMENTED** [review]
looks fine to me. with few exceptions:
-the Color class doesn't do much as of now. a color in this scenario is also a range within the three bands (R[]G[]B[]). you're working with the hardware so there's no predetermined color values to speak of since the readout is fluctuating constantly (hence the averaging function)
- I know u meant it as a generic function but it's not within a generic context, so "freq" instead of "v" would suggest the role of the code better

**Review by slatan1983 — APPROVED** [review]

**Pull request merged**